### PR TITLE
tailscale: add `contacts` resource

### PR DESCRIPTION
**Pull request by mpminardi**
Add a `tailscale_contacts` resource to allow for managing tailscale contacts settings via Terraform.

Updates https://github.com/tailscale/corp/issues/21631

### CODE REVIEW

File: tailscale/resource_contacts.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+func resourceContactsRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*tailscale.Client)
+
+	contacts, err := client.Contacts(ctx)
+	if err != nil {
+		return diagnosticsError(err, "Failed to fetch contacts")
+	}
+
+	accountMap := []map[string]interface{}{
```
Comment (oxtoacart) [inline tailscale/resource_contacts.go:111]: These feel somewhat repetitive, could maybe be pulled into a function like `buildContact()` or something?

File: tailscale/resource_contacts.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+	}
+
+	return nil
+}
+
+func resourceContactsUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*tailscale.Client)
+
+	if d.HasChange("account") {
+		account := d.Get("account").(*schema.Set).List()
```
Comment (oxtoacart) [inline tailscale/resource_contacts.go:145]: These sections feel a little repetitive, maybe they could be pulled into a helper function?

**Review by oxtoacart — APPROVED** [review]
Just a couple of suggestions about maybe making it a little more DRY, otherwise LGTM. Nice job on the documentation piece!

File: tailscale/resource_contacts.go (hunk 2 above)
Comment (mpminardi) [reply tailscale/resource_contacts.go]: Made an `updateContact` helper to encapsulate some of the repetitive bits here.

File: tailscale/resource_contacts.go (hunk 1 above)
Comment (mpminardi) [reply tailscale/resource_contacts.go]: Have added a `buildContactMap` function to clean this up a bit.

**Pull request merged**